src/Compilers/Core/Portable/SourceGeneration/AdditionalSourcesCollection.cs

```diff
@@ -84,11 +83,6 @@ public void Add(string hintName, SourceText source)
                 throw new ArgumentException(string.Format(CodeAnalysisResources.SourceTextRequiresEncoding, hintName), nameof(source));
             }
 
-            if (Path.IsPathRooted(hintName) || !Path.GetFullPath(hintName).StartsWith(Environment.CurrentDirectory, _hintNameComparison))
```

Comment:
We could potentially keep the `Path.IsPathRooted` check if we think its worth it?

Reply:
If we keep it though and fire and error then we potentially break existing code. 😦